File: system-test/sec-tests/typescript/sec_48_zip_slip.ts
// SEC-48: Zip Slip / Archive Extraction
// Test: TC-CR-052
import AdmZip from 'adm-zip';

function extractZip(filePath: string): void {
  const zip = new AdmZip(filePath);
  zip.extractAllTo('/tmp/extract/', true);  // No path traversal check
}

File: system-test/sec-tests/java/Sec48ZipSlip.java
// SEC-48: Zip Slip / Archive Extraction
// Test: TC-CR-052
package com.example;

import java.io.*;
import java.util.zip.*;

public class ZipService {

    public void extractZip(InputStream inputStream) throws Exception {
        ZipInputStream zis = new ZipInputStream(inputStream);
        ZipEntry entry;
        while ((entry = zis.getNextEntry()) != null) {
            // No path traversal check — allows ../ in entry names
            File outputFile = new File("/tmp/extract/" + entry.getName());
            if (!entry.isDirectory()) {
                try (FileOutputStream fos = new FileOutputStream(outputFile)) {
                    byte[] buffer = new byte[1024];
                    int len;
                    while ((len = zis.read(buffer)) > 0) {
                        fos.write(buffer, 0, len);
                    }
                }
            }
        }
    }
}
